apollo-router/src/services/router/service.rs

```diff
@@ -679,7 +679,7 @@ struct TranslateError<'a> {
 }
 
 // Process the headers to make sure that `VARY` is set correctly
-fn process_vary_header(headers: &mut HeaderMap<HeaderValue>) {
+pub(crate) fn process_vary_header(headers: &mut HeaderMap<HeaderValue>) {
```

Comment:
I think that we can migrate our tests into files without requiring us to change the visibility of the service functionality.

For example:
```
garypen@Garys-MacBook-Pro router % git diff
diff --git a/apollo-router/src/services/router.rs b/apollo-router/src/services/router.rs
index 367ba7042..f2df3278c 100644
--- a/apollo-router/src/services/router.rs
+++ b/apollo-router/src/services/router.rs
@@ -34,8 +34,6 @@ pub type Body = hyper::Body;
 pub type Error = hyper::Error;
 
 pub(crate) mod service;
-#[cfg(test)]
-mod tests;
 
 assert_impl_all!(Request: Send);
 /// Represents the router processing step of the processing pipeline.
diff --git a/apollo-router/src/services/router/service.rs b/apollo-router/src/services/router/service.rs
index 8f6ce0001..e29f6cd9f 100644
--- a/apollo-router/src/services/router/service.rs
+++ b/apollo-router/src/services/router/service.rs
@@ -679,7 +679,7 @@ struct TranslateError<'a> {
 }
 
 // Process the headers to make sure that `VARY` is set correctly
-pub(crate) fn process_vary_header(headers: &mut HeaderMap<HeaderValue>) {
+fn process_vary_header(headers: &mut HeaderMap<HeaderValue>) {
     if headers.get(VARY).is_none() {
         // We don't have a VARY header, add one with value "origin"
         headers.insert(VARY, ORIGIN_HEADER_VALUE.clone());
@@ -790,3 +790,7 @@ impl RouterCreator {
         self.supergraph_creator.planner()
     }
 }
+
+#[cfg(test)]
+#[path = "tests.rs"]
+mod tests;
diff --git a/apollo-router/src/services/router/tests.rs b/apollo-router/src/services/router/tests.rs
index 8a44206af..6baf55488 100644
--- a/apollo-router/src/services/router/tests.rs
+++ b/apollo-router/src/services/router/tests.rs
@@ -13,10 +13,9 @@ use serde_json_bytes::json;
 use tower::ServiceExt;
 use tower_service::Service;
 
+use super::*;
 use crate::graphql;
 use crate::services::router;
-use crate::services::router::service::from_supergraph_mock_callback;
-use crate::services::router::service::process_vary_header;
 use crate::services::subgraph;
 use crate::services::supergraph;
 use crate::services::SupergraphRequest;
garypen@Garys-MacBook-Pro router % 
```

I think that's neater.